cairo/src/contracts/token/extensions/hyp_erc721_URI_storage.cairo

```diff
@@ -1,48 +1,213 @@
+    #[abi(embed_v0)]
+    impl GasRouterImpl = GasRouterComponent::GasRouterImpl<ContractState>;
+
+    // TokenRouter
+    #[abi(embed_v0)]
+    impl TokenRouterImpl = TokenRouterComponent::TokenRouterImpl<ContractState>;
+    impl TokenRouterInternalImpl = TokenRouterComponent::TokenRouterInternalImpl<ContractState>;
+
+    //HypERC721
+
+    impl HypErc721Impl = HypErc721Component::HypErc721Impl<ContractState>;
+    impl HypErc721InternalImpl = HypErc721Component::HypErc721InternalImpl<ContractState>;
+
+    //ERC721
+    #[abi(embed_v0)]
+    impl ERC721MixinImpl = ERC721Component::ERC721MixinImpl<ContractState>;
+    impl ERC721InternalImpl = ERC721Component::InternalImpl<ContractState>;
+
+    impl UpgradeableInternalImpl = UpgradeableComponent::InternalImpl<ContractState>;
+
     #[storage]
     struct Storage {
-        mailbox: u256,
+        #[substorage(v0)]
+        ownable: OwnableComponent::Storage,
+        #[substorage(v0)]
+        mailboxclient: MailboxclientComponent::Storage,
+        #[substorage(v0)]
+        router: RouterComponent::Storage,
+        #[substorage(v0)]
+        gas_router: GasRouterComponent::Storage,
+        #[substorage(v0)]
+        token_router: TokenRouterComponent::Storage,
+        #[substorage(v0)]
+        hyp_erc721: HypErc721Component::Storage,
+        #[substorage(v0)]
+        erc721: ERC721Component::Storage,
+        #[substorage(v0)]
+        src5: SRC5Component::Storage,
+        #[substorage(v0)]
+        upgradeable: UpgradeableComponent::Storage,
     }
 
-    fn constructor() {}
-
-    impl HypERC721URIStorageImpl of super::IHypERC721URIStorage<ContractState> {
-        fn initialize(ref self: ContractState) {}
+    #[event]
+    #[derive(Drop, starknet::Event)]
+    enum Event {
+        #[flat]
+        OwnableEvent: OwnableComponent::Event,
+        #[flat]
+        MailboxclientEvent: MailboxclientComponent::Event,
+        #[flat]
+        RouterEvent: RouterComponent::Event,
+        #[flat]
+        GasRouterEvent: GasRouterComponent::Event,
+        #[flat]
+        TokenRouterEvent: TokenRouterComponent::Event,
+        #[flat]
+        ERC721Event: ERC721Component::Event,
+        #[flat]
+        SRC5Event: SRC5Component::Event,
+        #[flat]
+        HypErc721Event: HypErc721Component::Event,
+        #[flat]
+        UpgradeableEvent: UpgradeableComponent::Event,
+    }
 
-        fn balance_of(self: @ContractState, account: u256) -> u256 {
-            0
-        }
+    #[constructor]
+    fn constructor(
+        ref self: ContractState,
+        owner: ContractAddress,
+        mailbox: ContractAddress
+    ) {
+        self.ownable._transfer_ownership(owner);
+        self.mailboxclient.initialize(mailbox, Option::None, Option::None);
+    }
 
-        fn token_uri(self: @ContractState, token_id: u256) -> u256 {
-            0
+    #[abi(embed_v0)]
+    impl HypERC721URIStorageImpl of super::IHypERC721URIStorage<ContractState> {
+        fn initialize(
```

Comment:
Can you check the initialization patterns we used in other places. We can move these to constructor. Solidity version chosed this way cuz they are using proxies to have upgradeability

Reply:
okay, I will check and fix